### PR TITLE
feat(flow): adapt Flow for StarRouting

### PR DESCRIPTION
**Pull request by jacobowitz**
This PR closes https://github.com/jina-ai/internal-tasks/issues/274
This PR does:

- Adapt `Flows` to StarRouting
- Reenable all Flow unit tests (and delete some which are a bit phishy)
- Implement the use in K8s without the K8sGrpcConnectionPool
- Implement error handling for failing requests in the Executor
- @JoanFM implemented the plotting for star routing (https://github.com/jina-ai/jina/pull/3994) and merged it into this branc
- Fixing one issue with waiting for shutdown/ready in jinad

### CODE REVIEW

File: jina/proto/jina.proto [hunk 1]
```diff
@@ -140,10 +140,9 @@ message DocumentProto {
  */
 message RouteProto {
     string pod = 1; // the name of the BasePod
-    string pod_id = 2; // the id of the BasePod
-    google.protobuf.Timestamp start_time = 3; // receiving time
-    google.protobuf.Timestamp end_time = 4; // sending (out) time
-    StatusProto status = 5; // the status of the execution
+    google.protobuf.Timestamp start_time = 2; // receiving time
```
Comment (JoanFM) [inline jina/proto/jina.proto:143]: This is not accurate anymore, we should explicitly define what a route means. Sending from gateway time, receiving in gateway time, etc ...

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by jacobowitz** [comment]
I just noticed that I broke disabling the k8s connection pool

File: daemon/api/dependencies.py [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
-from jina.helper import cached_property
-from jina.peapods.peas.helper import update_runtime_cls
-from jina.peapods.peas.container_helper import get_gpu_device_requests
+from jina import Flow
 from jina.enums import (
-    PeaRoleType,
-    SocketType,
     RemoteWorkspaceState,
 )
+from jina.helper import cached_property
```
Comment (JoanFM) [inline daemon/api/dependencies.py:15]: change to relative dependencies

File: jina/flow/base.py [hunk 3]
```diff
@@ -1454,10 +1527,14 @@ def protocol(self) -> GatewayProtocolType:
...
     @protocol.setter
     def protocol(self, value: Union[str, GatewayProtocolType]):
-        """Set the protocol of this Flow
+        """Set the protocol of this Flow, can only be set before the Flow has been started
 
         :param value: the protocol to set
         """
+        # Flow is running already, protocol cant be changed anymore
+        if self._build_level >= FlowBuildLevel.RUNNING:
+            raise RuntimeError("Protocol can not be changed after the Flow has started")
```
Comment (JoanFM) [inline jina/flow/base.py:1536]: single quote

**Review by JoanFM — CHANGES_REQUESTED** [review]
Really small things

File: daemon/api/dependencies.py (hunk 2 above)
Comment (jacobowitz) [reply daemon/api/dependencies.py:15]: I think thats not possible here as its not in the jina folder but in the daemon folder?

**Pull request merged**